src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import {
+  Input as CoreInput,
+  InputProps as CoreInputProps,
+} from 'wix-ui-core/input';
+import style from './TextField.st.css';
+import { ErrorProps } from '../ErrorMessageWrapper';
+import { ReactComponent as ErrorIcon } from '../../assets/icons/Error.svg';
+import { Tooltip } from '../Tooltip';
+import { TooltipSkin } from '../Tooltip/TooltipEnums';
+import { TextFieldTheme } from './TextFieldEnums';
+import { EMPTY, ERROR, ERROR_MESSAGE, SUCCESS, THEME } from './dataKeys';
+
+export interface TPATextFieldProps {
+  /** the error message to display */
+  errorMessage?: string;
+  /** possible values: 'line', 'box' */
+  theme?: TextFieldTheme;
+  /** apply success state */
+  success?: boolean;
+  /** apply error state*/
+  error?: boolean;
+}
+export type TextFieldProps = ErrorProps & TPATextFieldProps & CoreInputProps;
+
+export const TextField: React.FunctionComponent<TextFieldProps> = (
+  props: TextFieldProps,
+) => {
+  const {
+    errorMessage,
+    success = false,
+    error = false,
+    disabled = false,
+    theme = TextFieldTheme.Box,
+    suffix,
+    ...coreInputProps
+  } = props;
+  let suffixToShow = suffix;
+
+  if (errorMessage && error) {
+    suffixToShow = (
+      <Tooltip
+        appendTo="scrollParent"
+        placement="top-end"
+        skin={TooltipSkin.Error}
+        content={errorMessage}
+      >
+        <ErrorIcon />
+      </Tooltip>
+    );
+  }
+  const dataObject = {
+    [ERROR_MESSAGE]: errorMessage,
+    [THEME]: theme,
+    [SUCCESS]: success,
+    [ERROR]: error,
+    [EMPTY]: !props.value,
+  };
+  const test = (props as any).test;
+  console.info(test);
+  return (
+    <CoreInput
+      {...dataObject}
+      {...style('root', {
+        test, //needed for visual test. hides text caret removes transitions
```

Comment:
Can you explain the `test` prop?
If it's only for hiding the caret, I think we can configure applitools for it.
We prefer to not add production code that's used only for testing.

Reply:
It also removes the `transition` from `border`. We can configure the Aplitools to handle both text caret and transitions?

Reply:
I think we can configure the test to ignore the caret.
Regarding transitions, we can add a short timeout before the snapshot for this.
Nevertheless, this code should be removed.

Reply:
removed.